src/core/SIP/SIPParameters.cs

```diff
@@ -259,13 +260,13 @@ public void Remove(string name)
         {
             if (name != null)
             {
-                m_dictionary.Remove(name);
+                m_dictionary.TryRemove(name, out string ignore);
             }
         }
 
         public void RemoveAll()
         {
-            m_dictionary = new Dictionary<string, string>();
+            m_dictionary = new ConcurrentDictionary<string, string>();

```

Comment:
Instead of instantiating a new instance keep the existing one and clear it:

```suggestion
            m_dictionary.Clear();
```